kolibri/utils/main.py

```diff
@@ -236,6 +236,10 @@ def _post_django_initialization():
                 settings.CACHES["process_cache"]["TIMEOUT"],
                 **settings.CACHES["process_cache"]["OPTIONS"]
             )
+        except AttributeError:
```

Comment:
I guess the other way of handling this would be to check if the `process_cache` is not the default cache - which in the case of Redis it is, and I assume for online instances also?

Reply:
I don't understand this comment. If `process_cache` is in `CACHES` (i.e. it's not the `default` cache) it means we're not using redis.
The logic would be mostly the same. We can remove the attribute error checking the cache BACKEND. Is that what you have in mind? :

```

def _post_django_initialization():
    from kolibri.deployment.default.cache import CACHES

    if 'process_cache' in CACHES:
        backend = CACHES['process_cache']['BACKEND']
        if backend != 'django.core.cache.backends.db.DatabaseCache':
            try:
                process_cache.cull()
            except SQLite3DatabaseError:
                shutil.rmtree(process_cache.directory, ignore_errors=True)
                os.mkdir(process_cache.directory)
                process_cache._cache = FanoutCache(
                    process_cache.directory,
                    settings.CACHES["process_cache"]["SHARDS"],
                    settings.CACHES["process_cache"]["TIMEOUT"],
                    **settings.CACHES["process_cache"]["OPTIONS"]
                )


```

Reply:
Yeah, mostly just to make it based on Django settings, rather than our OPTIONS, so in the case that it's overridden by a custom settings file, we're still checking properly.

Reply:
ok, done it. I have also added a guard in `kolibri/core/device/migrations/0003_contentcachekey.py` to avoid the problems this migration had when running in a system using `DatabaseCache`.